File: app/src/main/java/com/app/rupiksha/adapters/FAQAdapter.java
package com.app.rupiksha.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;


import com.app.rupiksha.R;
import com.app.rupiksha.databinding.RowFaqListItemBinding;
import com.app.rupiksha.models.FaqModel;

import java.util.List;

public class FAQAdapter extends RecyclerView.Adapter<FAQAdapter.ViewHolder> {
    private Context context;
    List<FaqModel> faqModels;
    public FAQAdapter(Context context, List<FaqModel> faqList, OnItemClick onItemClick) {
        this.context = context;
        this.faqModels=faqList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RowFaqListItemBinding binding = DataBindingUtil.bind(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.row_faq_list_item, parent, false));
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
       holder.binding.tvFaqListTitle.setText(faqModels.get(position).getLabel());

        holder.binding.rvFaqList.setAdapter(new QuestionAnswerAdapter(context,faqModels.get(position).getTableData(),faqModels.get(position).getLabel(),new QuestionAnswerAdapter.OnItemClick() {
            @Override
            public void onClick(int position) {
                holder.binding.rvFaqList.getAdapter().notifyDataSetChanged();
            }
        }));
    }

    @Override
    public int getItemCount() {
        return faqModels.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        RowFaqListItemBinding binding;

        public ViewHolder(@NonNull RowFaqListItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    public static interface OnItemClick {
        public void onClick(int position);
    }
}

File: app/src/main/java/com/app/rupiksha/adapters/FAQAdapter.kt
package com.app.rupiksha.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.app.rupiksha.R
import com.app.rupiksha.databinding.RowFaqListItemBinding
import com.app.rupiksha.models.FaqModel

class FAQAdapter(
    private val context: Context,
    private val faqModels: List<FaqModel>,
    private val onItemClick: OnItemClick? = null
) : RecyclerView.Adapter<FAQAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding: RowFaqListItemBinding = DataBindingUtil.inflate(
            LayoutInflater.from(parent.context),
            R.layout.row_faq_list_item,
            parent,
            false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val model = faqModels[position]
        holder.binding.tvFaqListTitle.text = model.label

        model.tableData?.let { tableData ->
            holder.binding.rvFaqList.adapter = QuestionAnswerAdapter(
                context,
                tableData,
                model.label ?: "",
                object : QuestionAnswerAdapter.OnItemClick {
                    override fun onClick(pos: Int) {
                        holder.binding.rvFaqList.adapter?.notifyDataSetChanged()
                    }
                }
            )
        }
    }

    override fun getItemCount(): Int = faqModels.size

    inner class ViewHolder(val binding: RowFaqListItemBinding) : RecyclerView.ViewHolder(binding.root)

    interface OnItemClick {
        fun onClick(position: Int)
    }
}
